### PR TITLE
Outcome classes for promiser in files promises

### PR DESCRIPTION
**Pull request by vpodzime**
TODO:
- [x] adapt existing tests to the new functionality
- [x] add or extend existing tests for the new functionality

### CODE REVIEW

**Comment by vpodzime** [comment]
[Build Status](https://ci.cfengine.com/job/pr-pipeline/5313/)

**Comment by vpodzime** [comment]
[Build Status](https://ci.cfengine.com/job/pr-pipeline/5314/)

File: cf-agent/verify_files.c [hunk 1]
```diff
@@ -991,8 +991,46 @@ PromiseResult FindAndVerifyFilesPromises(EvalContext *ctx, const Promise *pp)
 
 /*****************************************************************************/
 
+static DefineClasses GetExpandedClassDefinitionConstraints(const EvalContext *ctx, const Promise *pp)
+{
+    const char *namespace = PromiseGetBundle(pp)->ns;
```
Comment (craigcomstock) [inline cf-agent/verify_files.c:996]: Should there be an `assert(pp != null);` here?

Comment (craigcomstock) [reply cf-agent/verify_files.c:996]: Ah, I see you do it down below in the exposed function and not in this static function. :+1:

**Pull request merged**